builtin/rebase.c

```diff
@@ -1066,25 +1088,26 @@ int cmd_rebase(int argc, const char **argv, const char *prefix)
 			strbuf_reset(&buf);
 			if (capture_command(&stash, &buf, GIT_MAX_HEXSZ))
 				die(_("Cannot autostash"));
+			trim_trailing_newline(&buf);
 			if (get_oid(buf.buf, &oid))
 				die(_("Unexpected stash response: '%s'"),
 				    buf.buf);
 			strbuf_reset(&buf);
 			strbuf_add_unique_abbrev(&buf, &oid, DEFAULT_ABBREV);
 
-			if (safe_create_leading_directories_const(
-					options.state_dir))
-				die(_("Could not create directory '%s'"),
+			if (safe_create_leading_directories_const(autostash))
+				die(_("Could not create directory for '%s'"),
```

Comment:
This should not be here, as it has nothing to do with the config setting but everything with --autostash. It should go into that commit.